statsmodels/survey/survey_model.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+
+class SurveyModel(object):
+
+    def __init__(self, design, model_class, init_args={}, fit_args={}):
+        self.design = design
+        self.model = model_class
+        self.init_args = dict(init_args)
+        self.fit_args = dict(fit_args)
+
+    def fit(self, y, X, replicates=None):
+        self.init_args["weights"] = self.design.weights
+        self.params = self._get_params(y, X)
+        if replicates is None:
```

Comment:
Let's implement a method `get_nrep` in `SurveyDesign` so we don't always have to go through this try/except block (it appears several time).